src/database/schema/ChatSchema.ts

```diff
@@ -250,6 +251,18 @@ class ChatSchema implements BaseDbSchema {
     let rawJson: string | null = null;
     let attachmentJson: string | null = null;
 
+    const isDeleted = ['notification-deleted', 'deleted'].includes(json?.message?.message_type);
+
+    let type = json?.message?.type;
+    if (isDeleted) {
+      type = MESSAGE_TYPE_DELETED;
+    }
+
+    let message = json?.message?.text || json?.message || '';
+    if (isDeleted) {
+      message = DELETED_MESSAGE_TEXT;
+    }
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for checking if a message is deleted and setting the `type` and `message` variables accordingly is duplicated in this hunk and in lines 300-311. Consider extracting this logic into a separate function to adhere to the DRY (Don't Repeat Yourself) principle.



<!-- This is an auto-generated comment by OSS CodeRabbit -->